resources/assets/js/components/ActivityFeed.tsx

```diff
@@ -0,0 +1,110 @@
+import Activity from "./Activity";
+import { RootState } from "../store/store";
+import { Comment } from "../models/types";
+import { DispatchType } from "../configureStore";
+import { fetchComments } from "../store/Job/jobActions";
+import { getComments, sortComments } from "../store/Job/jobSelector";
+import { commentTypeMessages } from "./CommentForm";
+
+interface ActivityFeedProps {
+  jobId: number;
+  comments: Comment[];
+  handleFetchComments: (jobId: number) => Promise<void>;
+}
+
+const ActivityFeed: React.FunctionComponent<ActivityFeedProps> = ({
+  jobId,
+  comments,
+  handleFetchComments,
+}) => {
+  useEffect((): void => {
+    handleFetchComments(jobId);
+  }, [handleFetchComments, jobId]);
+  const intl = useIntl();
+  const userName = (userId: number) => "getUserName()";
+  const userRole = (userId: number) => "getUserRole()";
+  const commentType = (type: number | null): string => {
+    switch (type) {
+      case 1:
+        return intl.formatMessage(commentTypeMessages.question);
+      case 2:
+        return intl.formatMessage(commentTypeMessages.recommendation);
+      case 3:
+        return intl.formatMessage(commentTypeMessages.requiredAction);
+      default:
+        return intl.formatMessage(commentTypeMessages.comment);
+    }
+  };
+  return (
+    <section data-c-padding="top(1)">
+      <h3 data-c-font-size="h3" data-c-color="stop" data-c-margin="bottom(1)">
+        <FormattedMessage
+          id="activityfeed.title"
+          defaultMessage="Previous Activity"
+          description="Title of activity feed."
+        />
+      </h3>
+
+      {comments.length !== 0 ? (
+        comments.map(
+          (comment): React.ReactElement => (
+            <Activity
+              key={comment.id}
+              name={userName(comment.user_id)}
+              userRole={userRole(comment.user_id)}
+              comment={comment.comment}
+              location={comment.location}
+              time={comment.created_at}
+              type={commentType(comment.type_id)}
+              link={{ url: "/", title: "", text: "" }}
+            />
+          ),
+        )
+      ) : (
+        <div
+          data-c-container="form"
+          data-c-padding="top(triple) bottom(triple)"
+        >
+          <div
+            data-c-background="white(100)"
+            data-c-card
+            data-c-padding="all(double)"
+            data-c-radius="rounded"
+            data-c-align="base(centre)"
+          >
+            <p>
+              <FormattedMessage
+                id="activityfeed.loading"
+                defaultMessage="Your activities are loading..."
+                description="Message indicating that the activity feed is still being loaded."
+              />
+            </p>
+          </div>
+        </div>
+      )}
+    </section>
+  );
+};
+
+const mapStateToProps = (
+  state: RootState,
+): {
+  comments: Comment[];
+} => ({
+  comments: sortComments(getComments(state)),
```

Comment:
ActivityFeed needs a way to filter comments (eg to show only comments associated with a particular location). I think it makes sense to pass a filter function as a prop, and simply call `comments.filter(filter)`. Your choice whether to do that here, before sorting, or in the ActivityFeed component.